regions/core/pixcoord.py

```diff
@@ -37,19 +37,11 @@ def __init__(self, x, y):
         self.x = self._standardize_coordinate(x)
         self.y = self._standardize_coordinate(y)
 
-        for a, b in ((x, y), (y, x)):
-            z, shape = a, np.asanyarray(a).shape
-            if shape == () or shape == (1,):
-                while np.asanyarray(z).shape != np.asanyarray(b).shape:
-                    z = np.append(z, a)
-
-                if self.x is a:
-                    self.x = z
-                else:
-                    self.y = z
-
         if np.asanyarray(self.x).shape != np.asanyarray(self.y).shape:
```

Comment:
You still have these `np.asanyarray` calls which will create two Numpy arrays unnecessarily in the scalar case.
Can't you just move the
```
x, y = np.broadcast_arrays(x, y)
```
at the top and get rid of this if and try except?

Reply:
Then the scalars will also be converted to numpy arrays, which we do not want.

Reply:
We can get rid of the try except if you are ok with default ValueError exception of `numpy.broadcast_arrays`

Reply:
Yes, I would suggest to just let the ValueError from Numpy go to the user, not catch and re-throw another ValueError. And in the test I would just put this:
```
    with pytest.raises(ValueError):
        PixCoord(np.array([1, 2]), [3, 4, 5, 6])
```
(the exception message could change in future Numpy versions, and is not terribly important, so best to not assert on that message string).

Reply:
I locally tried removing the if and putting the broadcast at the front:
```
     def __init__(self, x, y):
-
+        x, y = np.broadcast_arrays(x, y)
         self.x = self._standardize_coordinate(x)
         self.y = self._standardize_coordinate(y)
 
-        if np.asanyarray(self.x).shape != np.asanyarray(self.y).shape:
-            try:
-                self.x, self.y = np.broadcast_arrays(self.x, self.y)
-            except ValueError:
-                raise ValueError('Input parameters x and y cannot be broadcast')
-
```
The only fails that occur are in repr / str because that leads to floats for int inputs, where before apparently we preserved int:
```
E       AssertionError: assert 'Region:Point...(x=3.0,y=4.0)' == 'Region:PointP...oord(x=3,y=4)'
E           Region:PointPixelRegion
E         - center:PixCoord(x=3.0,y=4.0)
E         ?                    --    --
E         + center:PixCoord(x=3,y=4)
```
Hmm ... not sure what to do.

But it's a minor point, this PR is basically ready to go.

Reply:
Looks like SkyCoord also always goes to float64 for int inputs?

Reply:
So if SkyCoord really also always goes to float64:
```
In [19]: SkyCoord(np.array([3, 4]), np.array([4, 5]), unit='deg').data.lon.dtype
Out[19]: dtype('float64')
```
then I'd suggest to do the same for PixCoord here and to update the ~ 5 assert statements on repr / str tests to pass.

@sushobhana - What do you think?

Reply:
seems like SkyCoord always goes for float.

but the above case gets converted to float because when you broadcast scalar values , it returns 0d numpy arrays, which gets converted to float in `_standardize_coordinate()`

Reply:
If you want every scalar to get converted to float, then yes we can change the test cases for sure. I am cool with whatever you think!

Reply:
Just found this:
https://docs.scipy.org/doc/numpy/reference/generated/numpy.asscalar.html

So how about this for init?
```
x, y = np.broadcast_arrays(x, y)
if np.isscalar(x):
    x, y = np.asscalar(x), np.asscalar(y)
self.x = x
self.y = y
```
Of if we prefer to always make float, then this:
```
x, y = np.broadcast_arrays(x, y)
if np.isscalar(x):
    x, y = float(x), float(y)
self.x = x
self.y = y
```
On that second point I have no opinion, but I still think putting broadcast_arrays first is the simplest way to write this, no?

Reply:
`np.isscalar(np.array(2))` returns false. but if we change that to `x.shape == ()` for the first case , it would work. we can get rid of the `_standardize_coordinate()` also.

Reply:
Then let's do that shape check, and then this is ready. Thanks for your patience with my comments!